hw5_to_do_list/oop/js/components.js

```diff
@@ -0,0 +1,361 @@
+class Button extends Component {
+    constructor(props) {
+        super({ ...props, elementType: 'button', })
+        this.state = {
+            disabled: props.isDisabledByDefault ? 'disabled' : false,
+        }
+        this.element.onclick = props.onClick
+    }
+
+    updateDOM() {
+        this.element.disabled = this.state.disabled
+        if (this.props.text) {
+            this.children = [this.props.text]
+        }
+    }
+
+    disable() {
+        this.setState({ disabled: 'disabled' })
+    }
+
+    enable() {
+        this.setState({ disabled: false })
+    }
+}
+
+
+class List extends Component {
+    constructor(props) {
+        super({ ...props, elementType: 'ul' })
+        this.state = {
+            items: this.hasSavedState() ? this.retrieveFromLocalStorage() : []
+        }
+        this.title = new Header({ text: props.title, level: 2 })
+    }
+
+    updateDOM() {
+        this.saveToLocalStorage()
+        this.children = [
+            this.title,
+            ...this.state.items
+        ]
+    }
+
+    removeItem = item => {
+        let items = [...this.state.items]
+        const idx = items.findIndex(i => { i.rawItem == item.rawItem })
```

Comment:
```suggestion
        const idx = items.findIndex(i => i.rawItem === item.rawItem)
```
!Careful! Your findIndex always returned undefined.
2. it would be better to use `===` always